core/src/com/unciv/ui/overviewscreen/NotificationsOverviewTable.kt

```diff
@@ -0,0 +1,96 @@
+import com.unciv.ui.utils.onClick
+import com.unciv.ui.worldscreen.WorldScreen
+
+class NotificationsOverviewTable(
+    val worldScreen: WorldScreen,
+    viewingPlayer: CivilizationInfo,
+    overviewScreen: EmpireOverviewScreen
+) : EmpireOverviewTab(viewingPlayer, overviewScreen) {
+
+    val notificationLog = viewingPlayer.notificationsLog
+
+    private val notificationTable = Table(BaseScreen.skin)
+
+    val scaleFactor = 0.3f
+    val inverseScaleFactor = 1f / scaleFactor
+    private val maxWidthOfStage = 0.333f
+    private val maxEntryWidth = worldScreen.stage.width * maxWidthOfStage * inverseScaleFactor
+
+    val iconSize = 20f
+
+    init {
+        val tablePadding = 30f
+        defaults().pad(tablePadding).top()
+
+        generateNotificationTable()
+
+        add(notificationTable)
+    }
+
+    private fun generateNotificationTable() {
+        if (viewingPlayer.notifications.isNotEmpty())
+            notificationTable.add(notificationsArrayTable("Current", viewingPlayer.notifications)).row()
+
+        for (index in notificationLog.indices) {
+            val turnCounter = notificationLog.lastIndex - index
+            notificationTable.add(notificationsArrayTable(turnCounter.toString(), notificationLog[index].notifications))
+            notificationTable.padTop(20f).row()
+        }
+    }
+
+    private fun notificationsArrayTable(index: String, notifications: ArrayList<Notification>): Table {
+        val turnTable = Table(BaseScreen.skin)
+        if (index != "Current")
+            turnTable.add("Turn $index").row()
+        else
+            turnTable.add("Current turn").row()
+
+        for (index2 in notifications.indices) {
+            val notification = Table(BaseScreen.skin)
+
+            val labelWidth = maxEntryWidth * notifications[index2].icons.size - 10f
+            val label = WrappableLabel(notifications[index2].text, labelWidth, Color.BLACK, 20)
+
+            notification.add(label)
+            notification.background = ImageGetter.getRoundedEdgeRectangle()
+            notification.touchable = Touchable.enabled
+            notification.onClick {
+                UncivGame.Current.resetToWorldScreen()
+                notifications[index2].action?.execute(worldScreen)
+            }
+
+            if (notifications[index2].icons.isNotEmpty()) {
+                val ruleset = worldScreen.gameInfo.ruleSet
+                for (icon in notifications[index2].icons.reversed()) {
+                    val image: Actor = when {
+                        ruleset.technologies.containsKey(icon) -> ImageGetter.getTechIcon(icon)
+                        ruleset.nations.containsKey(icon) -> ImageGetter.getNationIndicator(
+                            ruleset.nations[icon]!!,
+                            iconSize
+                        )
+                        ruleset.units.containsKey(icon) -> ImageGetter.getUnitIcon(icon)
+                        else -> ImageGetter.getImage(icon)
```

Comment:
This looks like the same logic for the notifications in the main screen, so how about unifying them? This could be a function of Notification in fact - getIconImage(ruleset:Ruleset) or something similar